lib/Sema/CSSimplify.cpp

```diff
@@ -1139,9 +1139,29 @@ ConstraintSystem::matchFunctionTypes(FunctionType *func1, FunctionType *func2,
   // that expect a function that takes a single tuple (of the same
   // arity).
   auto isSingleParam = [&](ArrayRef<AnyFunctionType::Param> params) {
```

Comment:
Would it be better if this was a top-level static function now that it's a bit longer?

Reply:
Extracted together with `fixMissingArguments`